### PR TITLE
Adds pagination to lists

### PR DESCRIPTION
**Pull request by melegiul**
### Short description

Adds pagination for all objects lists

### Proposed changes

Generally shows pagination links only if there is more than one page to display, else hides all pagination links.
There is always a link to the first and the last page.
Additionally there are links for 2 previous and next pages.
To limit the number of links all other pages are displayed as `...`
Adds pagination to the following lists:

Admin:
- [x] Regions
- [x] Languages
- [x] Users
- [x] Roles
- [x] Organizations
- [x] Offer templates

Region:
- [x] Region user
- [x] Offers
- [x] Events
- [x] POIs
- [x] Push Notifications

### Lists without pagination
Trees and pagination should not be combined, therefore pagination is not applied to:
- Page tree
- Language tree

### Feedback lists and pagination

At the moment feedback is split into multiple querysets, which must first be combined to a single queryset, before pagination can be applied. So for now feedback pagination is skipped until the feedback refactoring is merged.

### Resolved issues

Fixes: #620, #666

### Screenshots

pagination1

pagination2

### CODE REVIEW

**Review by timoludwig — APPROVED** [review]
Very cool, I really like it! :rocket: 

I think you can already merge this PR and don't have to wait until the Feedback PR #669 is merged.

File: src/backend/settings.py
Comment (timoludwig) [inline src/backend/settings.py:533]: Oh, just one thought: Maybe 25 is a bit much? Since one table row has a height of ~50px, 25 rows result in a total height of 1250px. If we assume that most people have a full HD resolution with a screen height of 1080px, they cannot use the list without scrolling. I think 16-18 entries would maybe fit better?

File: src/cms/templates/pages/page_form.html
Comment (melegiul) [reply src/cms/templates/pages/page_form.html:257]: Yes, I encountered #666 when I was working on the pagination links and fixed it to increase my pages test set.

File: src/backend/settings.py
Comment (timoludwig) [inline src/backend/settings.py:533]:
```suggestion
#: Number of entries displayed per pagination chunk
#: see :class:`~django.core.paginator.Paginator`
PER_PAGE = 16
```
Oh, and yet another thing: If you use the syntax `#:` for docstrings, don't include a newline between the docstring in the comment, otherwise it will be ignored and the setting won't be contained in the documentation.

File: src/backend/settings.py
Comment (melegiul) [reply src/backend/settings.py]: Yes, sounds good. I was also thinking about customization options, but it is always good to start with a default value that suits at least.:+1:

File: src/cms/templates/pages/page_form.html
Comment (melegiul) [reply src/cms/templates/pages/page_form.html:257]: in the origin version it was like:
```
{% if page %}
// mirrored page stuff
{% endif %}
```
So do you want to hide the "mirrored page" option when a completely new page form is displayed?
If yes I could add another condition next to expert mode in line 240.

Comment (timoludwig) [reply src/cms/templates/pages/page_form.html:257]: Oh, I just noticed that there is an `Internal Server Error` if a mirrored region is selected in the new page form. So until this is fixed, it's better to hide this field, I think.

**Comment by melegiul** [comment]
> Very cool, I really like it! rocket
> 
> I think you can already merge this PR and don't have to wait until the Feedback PR #669 is merged.

Thank you very much, I have to take a look also soon at our new feedback system.:rocket:
When the feedback is merged, we need only a few lines of code, to extend them with pagination.

File: src/cms/templates/pages/page_form.html
Comment (timoludwig) [reply src/cms/templates/pages/page_form.html:257]: However, this isn't really part of this PR anyway, so I will open a new issue.

Comment (melegiul) [reply src/cms/templates/pages/page_form.html:257]: Thanks, I triggered the merge too soon, friday evening :sweat_smile:

**Pull request merged**
